test/gtest/kernel_tuning_net.cpp

```diff
@@ -0,0 +1,99 @@
+    tensor<float> test_outputs_b = tensor<float>(miopenFloat, miopenTensorNCHW, 512, 288, 56, 56);
+
+    tensor<float> test_inputs_fallback  = tensor<float>(miopenFloat, miopenTensorNCHW, 1, 4, 2, 2);
+    tensor<float> test_weights_fallback = tensor<float>(miopenFloat, miopenTensorNCHW, 4, 4, 1, 1);
+    tensor<float> test_outputs_fallback = tensor<float>(miopenFloat, miopenTensorNCHW, 1, 4, 2, 2);
+
+    miopen::ConvolutionDescriptor conv_desc_fallback{2,
+                                                     miopenConvolution,
+                                                     miopenPaddingDefault,
+                                                     std::vector<int>{0, 0},
+                                                     std::vector<int>{1, 1},
+                                                     std::vector<int>{1, 1},
+                                                     std::vector<int>{0, 0},
+                                                     1,
+                                                     float(1)};
+
+    miopen::ConvolutionDescriptor conv_desc{std::vector<int>{0, 0},
+                                            std::vector<int>{1, 1},
+                                            std::vector<int>{1, 1},
+                                            std::vector<int>{0, 0},
+                                            1,
+                                            float(1)};
+
+    miopen::ProblemDescription forward_pd(test_inputs_f.desc,
+                                          test_weights_f.desc,
+                                          test_outputs_f.desc,
+                                          conv_desc,
+                                          miopen::conv::Direction::Forward);
+
+    miopen::ConvolutionContext forward(forward_pd);
+
+    forward.SetStream(&handle);
+    forward.DetectRocm();
+
+    miopen::ProblemDescription backward_pd(test_inputs_b.desc,
+                                           test_weights_b.desc,
+                                           test_outputs_b.desc,
+                                           conv_desc,
+                                           miopen::conv::Direction::BackwardData);
+
+    miopen::ConvolutionContext backward(backward_pd);
+
+    backward.SetStream(&handle);
+    backward.DetectRocm();
+
+    miopen::ProblemDescription fallback_pd(test_inputs_fallback.desc,
+                                           test_weights_fallback.desc,
+                                           test_outputs_fallback.desc,
+                                           conv_desc_fallback,
+                                           miopen::conv::Direction::Forward);
+
+    miopen::ConvolutionContext fallback(fallback_pd);
+
+    fallback.SetStream(&handle);
+    fallback.DetectRocm();
+
+    miopen::solver::PerformanceConfigConvAsm1x1U config_forward;
+    config_forward.HeuristicInit(forward, forward_pd);
```

Comment:
`HeuristicInit()` is not part of the public API and is only used for default config initialization. It was not even implemented for some solvers. You can use `GetDefaultPerformanceConfig()` for this purpose.

Reply:
@averinevg Well, it's ok to use internal APIs in the unit tests. However, yes, I would recommend instantiating the solver and taking default tuning config from it.

Reply:
I'm going to keep it as it is, don't see any valid reason to change it. This is the function that is the caller for all the other functions I changed in this PR so I want to test only what I've changed not anything else.

Reply:
I thought that the ultimate goal is to speed up getting a valid config and improve the library as a whole.

Reply:
Correct which can be seen through the performance testing files that I've attached to this PR. This test is just to make sure that my model is producing the expected outputs and that it is being feed the correct convolution parameters in the properly encoded manner as well as adding the ability to check for valid partial parameter sequences which ensures the model does not output invalid parameters. All three of these things are tested by just calling the HeuristicInit() function with these golden values and checking for an output of exact parameter sequences as this pre-trained model is deterministic.

Reply:
The performance of the model will not change unless the perfdb entries for ConvAsm1x1U change or need to be updated. At this point we will get this information from the Tuna team and then we would need to completely retrain the model on this new data for ConvAsm1x1U if there is substantial change to it's perfdb entries.

Reply:
@Dmantri98 Let's create a ticket for this task (smoke test would be enough I believe) and resolve this review comment. I recommend adding a link to this comment thread to the new ticket.